### PR TITLE
add costPaid and costOwed to activityCost and modify Person to support these

### LINKED ISSUE #108: Add costPaid and costOwed to ActivityCost to track them separately
e.g case 1, Payer involved
Ivan bought 5 milo packets, costing each $1 for all of us
Activity costs:
Ivan, $5, $1
Warren $0, $1
... (3 more entries of $0, $1)

e.g. case 2, Payer uninvolved
I take away 3 dollars chicken rice for you and Roy,
Activity costs:
Warren, $6, $0
Saurav, $0, $3
Roy, $0, $3

### PR DESCRIPTION
**Pull request by matheril**
ActivityCost now supports these constructors:
public ActivityCost(int activityId, double costPaid, double costOwed)
public ActivityCost(int activityId)
public ActivityCost() 

getters and setters for cost have been removed.
getters and setters for costPaid and costOwed have been added.

Person class' addActivityCost method has been replaced with
public void addActivityCost(int activityId, double costPaid, double costOwed)

Person class now has these three methods:
getTotalCostOwed() : returns the sum of all costOweds associated with this Person.
getTotalCostPaid() : returns the sum of all costPaids associated with this Person.
getTotalCost() : returns the sum of all costPaids - sum of all costOweds associated with this Person. This represents the amount of money everyone else owes to this Person.

Fixes #108

### CODE REVIEW

File: src/main/java/seedu/duke/data/Person.java [hunk 1]
```diff
@@ -27,17 +26,44 @@ public Person(String name) {
...
+     */
     public double getTotalCost() {
-        return totalCost;
+        double totalCostPaid = 0;
+        double totalCostOwed = 0;
+        for (ActivityCost i : activityCostList) {
+            totalCostPaid += i.getCostPaid();
+            totalCostOwed += i.getCostOwed();
+        }
+        return totalCostPaid - totalCostOwed;
```
Comment (warrencxw) [inline src/main/java/seedu/duke/data/Person.java:66]: This method could be abstracted to use the two methods above instead.

**Review by warrencxw — APPROVED** [review]
Looks good to merge, but there is one method that could be abstracted if you want to, to simply code.

File: src/main/java/seedu/duke/data/Person.java [hunk 2]
```diff
@@ -27,17 +26,44 @@ public Person(String name) {
...
+        for (ActivityCost i : activityCostList) {
+            totalCostOwed += i.getCostOwed();
+        }
+        return totalCostOwed;
+    }
+
+    /**
+     * Calculates the amount of money other Persons owe to this Person.
+     * In other words, how much money this Person must receive in order to break even.
+     * @return Sum of all costPaids - sum of all costOweds in the list of ActivityCosts for this Person.
```
Comment (ivanaitzliddat) [inline src/main/java/seedu/duke/data/Person.java:57]: Perhaps could add a new line at line 57

**Review by ivanaitzliddat — CHANGES_REQUESTED** [review]
Added comment for change

File: src/main/java/seedu/duke/data/Person.java (hunk 1 above)
Comment (matheril) [reply src/main/java/seedu/duke/data/Person.java:66]: Makes sense for readability but having to iterate through the list twice is a bit less efficient, so I will leave this unchanged for now.

File: src/main/java/seedu/duke/data/Person.java (hunk 2 above)
Comment (matheril) [reply src/main/java/seedu/duke/data/Person.java]: Resolved

**Review by ivanaitzliddat — APPROVED** [review]
Looks good to merge

**Pull request merged**